src/components/dougnutChart/DoughnutChart.tsx

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) 2022 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import React, { useEffect, useState, useRef } from "react";
+import "styles/components/DoughnutChart.scss"
+
+/**
+ * Props for the doughnut chart
+ * @param {size} is to get the width or height of the chart (Width and height would be equal in case of this chart). Defaults to 200px.
+ * @param {percentage} is to get the percentage of the chart
+ * @param {status} is to get the status color for the chart (Success, Warning and Danger). Defaults to "Success".
+ * @param {label} is to get the Label to be displayed in the chart.
+ * @param {showPercentage} is a boolean value for displaying percentage in label. Defaults to true.
+ * @param {trackWidth} is to get the width of the chart's track. Defaults to 15px.
+ * @param {labelLength} is an optional prop to get the truncation length of the label provided. Defaults to 15.
+ */
+
+export type CircularProgressType = {
+    size?: number;
```

Comment:
can we change `size, trackWidth and labelLength` as string ? so that user can specify other units like `rem/em/vh/vw` ?

Reply:
**Radius** and **Circumference** are calculated from the `size`  and `trackWidth` . Currently, we are getting these values as numbers. If we have to support different units like `rem/em/vh/vw` , we will have to get these values as strings and have to segregate the value and unit from the string for calculating radius and circumference and need to apply respective units for them. This requires additional effort and time.

Reply:
ok. Got it